planning/autoware_path_generator/src/node.cpp

```diff
@@ -0,0 +1,181 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "node.hpp"
+
+#include "autoware/path_generator/utils.hpp"
+
+#include <autoware_lanelet2_extension/utility/message_conversion.hpp>
+#include <autoware_lanelet2_extension/utility/route_checker.hpp>
+
+namespace autoware::path_generator
+{
+PathGenerator::PathGenerator(const rclcpp::NodeOptions & node_options)
+: Node("path_generator", node_options)
+{
+  param_listener_ =
+    std::make_shared<::path_generator::ParamListener>(this->get_node_parameters_interface());
+
+  // publisher
+  path_publisher_ = create_publisher<PathWithLaneId>("~/output/path", 1);
+
+  {
+    const auto planning_hz = declare_parameter<double>("planning_hz");
+    timer_ = rclcpp::create_timer(
+      this, get_clock(), rclcpp::Rate(planning_hz).period(), std::bind(&PathGenerator::run, this));
+  }
+}
+
+void PathGenerator::run()
+{
+  const auto input_data = takeData();
```

Comment:
Each variable in input_data may be nullptr. We should check whether the input_data is valid or not (= whether the input data is ready or not) here.

Reply:
The validation is now in the updatePlannerData.
I feel checking whether the input_data is valid or not should be before planPath